Dockerfile

```diff
@@ -1,100 +1,79 @@
-      apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 \
-      --recv-keys 421C365BD9FF1F717815A3895523BAEEB01FA116; \
-    else \
       apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 \
       --keyserver-options http-proxy=$http_proxy \
       --recv-keys 421C365BD9FF1F717815A3895523BAEEB01FA116; \
     fi
 
-ENV ROS1_DISTRO melodic
-ENV ROS2_DISTRO bouncy
-ENV ROSDISTRO_INDEX_URL 'https://raw.githubusercontent.com/ros/rosdistro/master/index.yaml'
-
-# update latest package versions
-RUN apt-get update
-RUN rm -rf /etc/ros/rosdep/sources.list.d/20-default.list
-
 # install ROS2 dependencies
-RUN apt install -y \
-    build-essential \
-    cmake \
-    git \
-    python3-colcon-common-extensions \
-    python3-pip \
-    python-rosdep \
-    python3-vcstool \
-    wget
+RUN apt-get update && apt-get install -q -y \
+      build-essential \
+      cmake \
+      git \
+      python3-colcon-common-extensions \
+      python3-vcstool \
+      wget \
+    && rm -rf /var/lib/apt/lists/*
 
-# install Fast-RTPS dependencies
-RUN apt install --no-install-recommends -y \
-    libasio-dev \
-    libtinyxml2-dev
+# copy ros package repo
+ENV NAV2_WS /opt/nav2_ws
+RUN mkdir -p $NAV2_WS/src
+WORKDIR $NAV2_WS/src
+COPY ./ navigation2/
 
-# install map_server dependencies
-RUN apt-get install -y \
-    libsdl-image1.2 \
-    libsdl-image1.2-dev \
-    libsdl1.2debian \
-    libsdl1.2-dev
-
-# get the latest nightly ROS2 build -> ros2_ws/ros2_linux
-WORKDIR /ros2_ws
-RUN wget -nv -t 5 https://ci.ros2.org/view/packaging/job/packaging_linux/lastSuccessfulBuild/artifact/ws/ros2-package-linux-x86_64.tar.bz2
-RUN tar -xjf ros2-package-linux-x86_64.tar.bz2
-
-# clone navigation2 repo
-WORKDIR /ros2_ws/navigation2_ws/src
-ARG USER=ros-planning
-RUN git clone https://github.com/$USER/navigation2.git
-
-# change to correct branch if $BRANCH is not = master
-WORKDIR /ros2_ws/navigation2_ws/src/navigation2
-ARG PULLREQ=false
-ARG BRANCH=master
-RUN echo "pullreq is $PULLREQ"
-RUN if [ "$PULLREQ" == "false" ] && [ "$BRANCH" == "master" ]; then \ 
-      echo "No pull request number given - defaulting to master branch"; \
-    elif [ "$BRANCH" != "master" ]; then \
-      cd navigation2; \
-      git fetch origin $BRANCH:temp_branch; \
-      git checkout temp_branch; \
-      cd -; \
-      echo "No pull request number given - defaulting to $BRANCH branch"; \
-    else \
-      git fetch origin pull/$PULLREQ/head:pr_branch; \
-      git checkout pr_branch; \
-    fi
+# clone dependency package repos
+ENV ROS_WS /opt/ros_ws
+RUN mkdir -p $ROS_WS/src
+WORKDIR $ROS_WS
+RUN vcs import src < $NAV2_WS/src/navigation2/tools/ros2_dependencies.repos
 
-# Download dependencies
-RUN echo "Downloading the ROS 2 navstack dependencies workspace"
-WORKDIR /ros2_ws/navstack_dependencies_ws/src
-RUN vcs import . < /ros2_ws/navigation2_ws/src/navigation2/tools/ros2_dependencies.repos
+# install dependency package dependencies
+RUN . /opt/ros/$ROS_DISTRO/setup.sh && \
```

Comment:
What sets the $ROS_DISTRO env var?

Reply:
This is an environment variable set upstream in all ros related images from osrf, and enables downstream images to be more postable across distro releases, e.g. preventing folks from having to hardcode a dockerfile to the distro installed from the current base image.

https://github.com/osrf/docker_images/blob/ea0fdabf250f816203d058dca1df12802a9c44cc/ros2/nightly/nightly/Dockerfile#L45